server/server.go

```diff
@@ -1755,19 +1756,27 @@ func readInputGetV1(str string) (ast.Value, error) {
 	return ast.InterfaceToValue(input)
 }
 
-func readInputPostV1(r io.ReadCloser) (ast.Value, error) {
+func readInputPostV1(r *http.Request) (ast.Value, error) {
 
-	bs, err := ioutil.ReadAll(r)
+	bs, err := ioutil.ReadAll(r.Body)
 
 	if err != nil {
 		return nil, err
 	}
 
 	if len(bs) > 0 {
 
+		ct := r.Header.Get("Content-Type")
+
 		var request types.DataRequestV1
 
-		if err := util.UnmarshalJSON(bs, &request); err != nil {
+		// There is no standard for yaml mime-type so we just look for
```

Comment:
Let's use `util.Umarshal` if the content-type is NOT JSON. If the content-type is JSON, use `util.UnmarshalJSON`.